camel/embeddings/base.py

```diff
@@ -0,0 +1,53 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+from abc import ABC, abstractmethod
+from typing import List
+
+
+class BaseEmbedding(ABC):
+    r"""Abstract base class for text embedding functionalities.  """
+
+    @abstractmethod
+    def embed_texts(self, texts: List[str]) -> List[List[float]]:
```

Comment:
This base class is designed only for text embedding. For image embedding, we have two options: either create a separate base class specifically for image embedding or make the `embed` function's parameter dynamic. This would allow the program to dynamically select the appropriate overridden function during runtime.